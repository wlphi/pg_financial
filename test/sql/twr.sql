DROP EXTENSION IF EXISTS financial CASCADE;
CREATE EXTENSION financial;
SET extra_float_digits = -4;

SELECT twr(amt, val, ts)
FROM (VALUES
    (-1000::double precision, 1000::double precision, '2023-01-01'::timestamptz),
    (0::double precision, 1500::double precision, '2023-02-01')
) x(amt, val, ts);

SELECT twr(amt, val, ts)
FROM (VALUES
    (-1000::double precision, 1000::double precision, '2023-01-01'::timestamptz),
    (0::double precision, 800::double precision, '2023-02-01')
) x(amt, val, ts);

SELECT twr(amt, val, ts)
FROM (VALUES
    (-1000::double precision, 1000::double precision, '2023-01-01'::timestamptz),
    (0::double precision, 1000::double precision, '2023-02-01')
) x(amt, val, ts);

SELECT twr(amt, val, ts)
FROM (VALUES
    (-1000::double precision, 1000::double precision, '2023-01-01'::timestamptz),
    (0::double precision, 2500::double precision, '2023-02-01')
) x(amt, val, ts);

SELECT twr(amt, val, ts)
FROM (VALUES
    (-1000::double precision, 1000::double precision, '2023-01-01'::timestamptz)
) x(amt, val, ts);

SELECT twr(NULL::double precision, 1000::double precision, '2023-01-01'::timestamptz);
SELECT twr(1000::double precision, NULL::double precision, '2023-01-01'::timestamptz);
SELECT twr(1000::double precision, 1000::double precision, NULL::timestamptz);

SELECT twr(amt, val, ts)
FROM (VALUES
    (-1000::double precision, -1000::double precision, '2023-01-01'::timestamptz),
    (0::double precision, 1000::double precision, '2023-02-01')
) x(amt, val, ts);

SELECT twr(amt, val, ts)
FROM (VALUES
    (-1000::double precision, 0::double precision, '2023-01-01'::timestamptz),
    (0::double precision, 1000::double precision, '2023-02-01')
) x(amt, val, ts);

SELECT twr(amt, val, ts)
FROM (VALUES
    (-1000::double precision, 1000::double precision, '2023-01-01'::timestamptz),
    (0::double precision, 1001::double precision, '2023-02-01')
) x(amt, val, ts);

SELECT twr(amt, val, ts) OVER (ORDER BY ts)
FROM (VALUES
    (-1000::double precision, 1000::double precision, '2023-01-01'::timestamptz),
    (-500::double precision, 1600::double precision, '2023-02-01'),
    (200::double precision, 1500::double precision, '2023-03-01')
) x(amt, val, ts);
