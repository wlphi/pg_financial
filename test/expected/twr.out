test_name    | twr_result | test_result 
---------------+------------+-------------
 regression_case |     0.1010 | t
 same_day       |     0.1000 | t
 simple_case    |     0.2100 | t
 single_row     |            | t
 with_flows     |     0.2500 | t
 zero_value     |            | t
(6 rows)

-- Empty set test result
 twr 
-----
     
(1 row)

-- NULL test results
ERROR:  twr amount input cannot be NULL
ERROR:  twr value input cannot be NULL
ERROR:  twr timestamp input cannot be NULL
ERROR:  twr amount input cannot be NULL
ERROR:  twr amount input cannot be NULL
ERROR:  twr value input cannot be NULL
ERROR:  twr amount input cannot be NULL
NOTICE:  All NULL tests passed successfully
